.config/webpack-config-dev.js

```diff
@@ -0,0 +1,148 @@
+					// Process JS with Babel.
+					{
+						// Use a custom cache directory.
+						...loaders.js,
+						options: {
+							cacheDirectory: filePath( `node_modules/.cache/${ config.name }/babel-loader` ),
+						},
+					},
+					{
+						test: /\.scss$/,
+						use: [
+							require.resolve( 'style-loader' ),
+							// Process SASS into CSS.
+							loaders.css,
+							loaders.postcss,
+							loaders.sass,
+						],
+					},
+					// "file" loader makes sure any non-matching assets still get served.
+					// When you `import` an asset you get its filename.
+					loaders.file,
+				],
+			},
+		],
+	},
+
+	// Optimize output bundle.
+	optimization: {
+		nodeEnv: 'development',
+	},
+
+	// If any of the above properties conflict, the version from the passed-in config will be used.
+	...config,
+
+	// Allow config to override shared devServer properties.
+	devServer: {
+		headers: {
+			'Access-Control-Allow-Origin': '*',
+		},
+		hot: true,
+		stats: {
+			all: false,
+			assets: true,
+			colors: true,
+			errors: true,
+			performance: true,
+			timings: true,
+			warnings: true,
+		},
+		watchOptions: {
+			aggregateTimeout: 300,
+		},
+		...( config.devServer || {} ),
+	},
+
+	// Allow config to add plugins.
+	plugins: [
+		new webpack.HotModuleReplacementPlugin(),
+		...( config.plugins || [] ),
+	],
+} );
+
+
+const devServerUri = ( port, path ) => `http://localhost:${ port }/${ path }`;
+
+// cosnt devServerPort = devServerPort();
+// Clean up manifest on exit.
+onExit( () => {
+	config.map( project => filePath( project.path, 'build', 'asset-manifest.json' ) )
+	.forEach( path => {
+		try {
+			unlinkSync( path );
+		} catch ( e ) {
+			// Silently ignore unlinking errors: so long as the file is gone, that is good.
+		}
+	} );
+} );
+
+const buildTargets = config.map( ( { name, path, entry }, i ) => devConfig( {
+	name,
+	devServer: {
+		port: devServerPort() + i,
+	},
+	entry,
+	context: relPath( path ),
+	output: {
+		// Add /* filename */ comments to generated require()s in the output.
+		pathinfo: true,
+		path: filePath( path, 'build' ),
+		publicPath: devServerUri( devServerPort() + i, `${path}/build/` ),
+		filename: '[name].js',
+	},
+	plugins: [
+		// Generate a manifest file which contains a mapping of all asset filenames
+		// to their corresponding output file so that PHP can pick up their paths.
+		new ManifestPlugin( {
+			publicPath: devServerUri( devServerPort() + i, `${path}/build/` ),
+			fileName: 'asset-manifest.json',
+			writeToFileEmit: true,
+		} ),
+	],
+	externals: wpExternalScripts,
+} ) );
+
+// Permit this same config file to be used by a full CLI build or a theme- or
+// plugin-specific dev server instance: If run with `webpack --config=...` as
+// normal, export all build targets together as a Webpack multi-config array.
+// If a specific target is specified with `BUILD_TARGET=blocks` or
+// `BUILD_TARGET=theme`, only export that specific build target configuration.
+module.exports = buildTargets
```

Comment:
ha - credit goes to... not sure actually! This exact line of code exists in several projects I used for reference.

Reply:
This was me, I think -- there was (is?) a bug where hot reloading won't bind properly if you're running two configs off the same bundle, so I added this on the first Gutenberg project I did so that we could run two dev servers simultaneously using `concurrently`, rather than having to switch back and forth.